sw/host/opentitanlib/src/util/openocd.rs

```diff
@@ -173,6 +187,23 @@ impl OpenOcdServer {
         self.openocd_socket_stream.replace(Some(stream));
         String::from_utf8(answer).context("failed to parse OpenOCD response as UTF-8")
     }
+
+    /// Poll `addr` until it is bound and a socket can connect.
+    fn wait_for_socket<A: ToSocketAddrs>(addr: A, timeout: Duration) -> io::Result<TcpStream> {
```

Comment:
It would probably be better to sleep between calls to connect. Since each call can also take time, you could measure the time before and after the `connect` call and sleep so that you retry every 100ms for example. It's probably better if the timeout is configurable I would say.

Reply:
Thanks, I've added a (non-configurable) polling delay of 200ms, plus made the timeout configurable with a 3s default.

Does this sound okay?

Reply:
Oops, just read that you suggested 100ms. Updated.